src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.IsComplexNumber.cs

```diff
@@ -0,0 +1,83 @@
+        /// <remarks>
+        /// This method effectively computes <c><paramref name="destination" />[i] = <typeparamref name="T"/>.IsComplexNumber(<paramref name="x" />[i])</c>.
+        /// </remarks>
+        public static void IsComplexNumber<T>(ReadOnlySpan<T> x, Span<bool> destination)
+            where T : INumberBase<T>
+        {
+            if (!MayBeComplex<T>())
+            {
+                if (x.Length > destination.Length)
+                {
+                    ThrowHelper.ThrowArgument_DestinationTooShort();
+                }
+
+                destination.Slice(0, x.Length).Clear();
+                return;
+            }
+
+            InvokeSpanIntoSpan<T, IsComplexNumberOperator<T>>(x, destination);
+        }
+
+        /// <summary>Computes whether all of the values in the specified tensor are complex numbers.</summary>
+        /// <param name="x">The tensor, represented as a span.</param>
+        /// <returns>
+        /// <see langword="true"/> if all of the values in <paramref name="x"/> are complex numbers; otherwise, <see langword="false"/>.
+        /// If <paramref name="x"/> is empty, <see langword="false"/> is returned.
+        /// </returns>
+        public static bool IsComplexNumberAll<T>(ReadOnlySpan<T> x)
+            where T : INumberBase<T> =>
+            !x.IsEmpty &&
+            MayBeComplex<T>() && All<T, IsComplexNumberOperator<T>>(x);
+
+        /// <summary>Computes whether any of the values in the specified tensor is a complex number.</summary>
+        /// <param name="x">The tensor, represented as a span.</param>
+        /// <returns>
+        /// <see langword="true"/> if any of the values in <paramref name="x"/> is a complex number; otherwise, <see langword="false"/>.
+        /// If <paramref name="x"/> is empty, <see langword="false"/> is returned.
+        /// </returns>
+        public static bool IsComplexNumberAny<T>(ReadOnlySpan<T> x)
+            where T : INumberBase<T> =>
+            !x.IsEmpty &&
+            MayBeComplex<T>() && Any<T, IsComplexNumberOperator<T>>(x);
+
+        /// <summary>Gets whether any value could be complex.</summary>
+        private static bool MayBeComplex<T>() =>
+            typeof(T) != typeof(Half) && typeof(T) != typeof(float) && typeof(T) != typeof(double) && typeof(T) != typeof(NFloat) &&
+            typeof(T) != typeof(decimal) &&
+            typeof(T) != typeof(sbyte) && typeof(T) != typeof(byte) &&
+            typeof(T) != typeof(short) && typeof(T) != typeof(ushort) && typeof(T) != typeof(char) &&
+            typeof(T) != typeof(int) && typeof(T) != typeof(uint) &&
+            typeof(T) != typeof(long) && typeof(T) != typeof(ulong) &&
+            typeof(T) != typeof(nint) && typeof(T) != typeof(nuint) &&
+            typeof(T) != typeof(Int128) && typeof(T) != typeof(UInt128);
```

Comment:
We have this and similar checks in a few places. I wonder if we could come up with a good shareable name for them to avoid duplication.

Like all of this can be summed up as effectively `IsBuiltInBinaryInteger || IsBuiltInIeee754FloatingPoint` + `decimal`, for example.